instrumentation/rails/lib/opentelemetry/instrumentation/rails/patches/action_controller/metal.rb

```diff
@@ -14,6 +14,10 @@ module Metal
             def dispatch(name, request, response)
               rack_span = OpenTelemetry::Instrumentation::Rack.current_span
               rack_span.name = "#{self.class.name}##{name}" if rack_span.context.valid? && !request.env['action_dispatch.exception']
+
+              ::Rails.application.routes.router.recognize(request) do |route, _params|
```

Comment:
When I originally implemented this instrumentation I had some concerns regarding the overhead of running [.recognize](https://github.com/rails/rails/blob/main/actionpack/lib/action_dispatch/journey/router.rb#L65) a second time for each request to the application.  

I'd like to get a second reviewer to weigh in on it, and if they don't share my same concerns then I have no reason to block on this.

Reply:
Yes, I can understand your concern, I didn't find a better way to get the route here. Maybe we can add a configuration to enable that behavior?

Reply:
👍 I'd be pro-config-option

Reply:
Agreed, if we put this behind a config option that defaults to it being off I think that's a good compromise for bringing this functionality in.

Reply:
Great! I will work on it

Reply:
Hey team, I updated the code, please take a look